src/modules/auth/auth.service.ts

```diff
@@ -82,4 +89,54 @@ export class AuthService {
       role,
     };
   }
+
+  async forgotPassword({ email }: AuthForgotPasswordDto) {
+    const user = await this.authRepo.findOneBy({ email });
+    console.log(user);
+
+    if (user) {
+      const userAbout = await this.signUser(user.id, user.email);
+
+      const result = await this.forgotPasswordRepo.save({
+        user: user,
+        link: userAbout.token,
+      });
+
+      console.log(result);
+
+      const url = `http://localhost:3000/restore-password?token=${userAbout.token}`;
+
+      const transport = await this.mailService.sendMail({
+        to: email,
+        subject: 'GetJob Forgot password',
+        from: 'milkav06062003@gmail.com',
+        html: `<h1>Change password</h1><p>If you want to change password go to:</p><a href="${url}">${url}</a>`,
+      });
+
+      console.log(`Email successfully dispatched to ${email}`);
+      return transport;
+    }
+    return false;
+  }
+
+  async restorePassword({ password, token }: AuthRestorePasswordDto) {
+    const forgotPassword = await this.forgotPasswordRepo.findOne({
+      where: {
+        link: token,
+      },
+      relations: {
+        user: true,
+      },
+    });
+    const salt = genSaltSync(10);
```

Comment:
10 is magic number, never use magic number. move it to the constant and set appropriate name

Reply:
done